### PR TITLE
Authorizer Support

### PR DESCRIPTION
**Pull request by m4dc4p**
* Methods for adding & removing authorizers
* `shareOnBehalfOf` method for authorizer's to use for sharing
* Methods for listing authorizers and those who have made a client
an authorizer.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ Publishing
 
 The SDK can be published for plain Java and for Android. To publish to your local Maven repository, run the following command:
 
-$ gradlew :publish:plain:publishToMavenLocal :publish:android:publishToMavenLocal
+$ gradlew :publish:publishLocal
```
Comment (ericmann) [inline DEVELOPMENT.md:136]: Simpler, cleaner, I like it :-)

**Review by ericmann — APPROVED** [review]

**Review by gstro — COMMENTED** [review]
LGTM so far, but there are a few more things I'd like to see here:
- I think you called out that you don't have `revokeOnBehalfOf`
- Doc comments
- At least one end-to-end test showing an authorizer authorizing a 3rd client, and that client able to pull down records and successfully decrypt. That we we're getting the EAK changes exercised as well, not just the policy endpoints. Nice work!

**Review by gstro — APPROVED** [review]
Ha, I missed your last commit which exactly addressed everything I mentioned in my previous review. Nice work!

File: e3db/src/main/java/com/tozny/e3db/AuthorizerPolicy.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.util.UUID;
+
+/**
+ * Used for two purposes:
+ *
+ * <ul>
+ *   <li>When returned by {@link Client#getAuthorizedBy(ResultHandler)}, represents a writer that authorized
+ *   this client to share on its behalf.</li>
+ *   <li>When returned by {@link Client#getAuthorizers(ResultHandler)}, represents another client that this
```
Comment (gstro) [inline e3db/src/main/java/com/tozny/e3db/AuthorizerPolicy.java:31]: It's probably worth updating to the different response types to match the spec: `GrantedAuthorizerPolicy` and `ProxyAuthorizerPolicy`.

**Review by ericmann — APPROVED** [review]

File: e3db/src/main/java/com/tozny/e3db/AuthorizerPolicy.java (hunk 2 above)
Comment (m4dc4p) [reply e3db/src/main/java/com/tozny/e3db/AuthorizerPolicy.java:31]: After discussion I'll leave this as-is, tho it isn't ideal.

**Review by ericmann — APPROVED** [review]

**Pull request merged**